qiskit/assembler/disassemble.py

```diff
@@ -59,6 +61,30 @@ def _disassemble_circuit(qobj) -> CircuitModule:
     return CircuitModule((_experiments_to_circuits(qobj), run_config, user_qobj_header))
 
 
+def _qobj_to_circuit_cals(qobj, pulse_lib, param_pulses):
+    qobj_cals = qobj.config.calibrations.to_dict()["gates"]
+    converter = QobjToInstructionConverter(pulse_lib)
+
+    qc_cals = {}
+    for gate in qobj_cals:
+        config = (tuple(gate["qubits"]), tuple(gate["params"]))
+        cal = {
+            config: pulse.Schedule(
+                name="%s %s %s" % (gate["name"], str(gate["params"]), str(gate["qubits"]))
+            )
+        }
+        for instruction in gate["instructions"]:
+            schedule = (
+                converter.convert_parametric(PulseQobjInstruction.from_dict(instruction))
+                if "pulse_shape" in instruction and instruction["pulse_shape"] in param_pulses
+                else converter(PulseQobjInstruction.from_dict(instruction))
+            )
+            cal[config] = cal[config].insert(schedule.ch_start_time(), schedule)
+        qc_cals.setdefault(gate["name"], cal).update(cal)
```

Comment:
```suggestion
        qc_cals.setdefault(gate["name"], cal)
```
I don't think you have to also call `update`